app/src/main/cpp/src/components/transport_manager/src/bluetooth_le/ble_control_protocol.cc

```diff
@@ -0,0 +1,102 @@
+#include "transport_manager/bluetooth_le/ble_control_protocol.h"
+
+#include "utils/logger.h"
+#include "utils/jsoncpp_reader_wrapper.h"
+#include "utils/helpers.h"
+
+#include <map>
+#include <algorithm>
+
+namespace transport_manager {
+namespace transport_adapter {
+
+SDL_CREATE_LOG_VARIABLE("TransportManager")
+
+namespace {
+    std::map<BleProtolActions, std::string> mapActionNames = {
+    {BleProtolActions::ON_DEVICE_CONNECTED, "ON_DEVICE_CONNECTED"},
+    {BleProtolActions::ON_DEVICE_DISCONNECTED, "ON_DEVICE_DISCONNECTED"}};
+
+    const char cActionKey[] = "action";
+    const char cNameKey[] = "name";
+    const char cAddressKey[] = "address";
+    const char cParamsKey[] = "params";
+
+    std::string GeStrFromParams(const Json::Value& params, const char* key) {
+        if(params != Json::Value::null) {
+
+            auto str = params.get(key, Json::Value::null);
+            if(str != Json::Value::null) {
+                return str.asString();
+            }
+        }
+
+        return std::string();
+    }
+
+}
+
+BleProtolActions BleControlProtocol::GetMessageActionType(const MsgData& message) {
+    utils::JsonReader reader_;
+    Json::Value value;
+
+    if(reader_.parse((char*)(message.data()), &value)) {
+
+        auto action = value.get(cActionKey, Json::Value::null);
+
+        if(action != Json::Value::null) {
+            return GetActionFromString(action.asString());
+        }
+    }
+
+    return BleProtolActions::INVALID_ENUM;
+}
+
+BleProtolActions BleControlProtocol::GetActionFromString(const std::string& str) {
+    BleProtolActions result = BleProtolActions::INVALID_ENUM;
+
+    const auto findResult = std::find_if(mapActionNames.begin(), mapActionNames.end(),
+        [&str](const std::pair<BleProtolActions, std::string> &pair) {
+        return pair.second == str;});
+
+    if(findResult != mapActionNames.end()) {
+        result = findResult->first;
+    }    
+
+    return result;
+}
+
+std::string BleControlProtocol::GetActionString(const BleProtolActions action) {
+    auto it = mapActionNames.find(action);
+    return it != mapActionNames.end() ? it->second : "UNKNOWN";
+}
+
+std::string BleControlProtocol::GetAddress(const MsgData& message) {
+    utils::JsonReader reader_;
+    Json::Value value;
+    
+    if(reader_.parse((char*)(message.data()), &value)) {
+
+        auto params = value.get(cParamsKey, Json::Value::null);
+
+        return GeStrFromParams(params, cAddressKey);
+    }
+
+    return std::string();
+}
+
+std::string BleControlProtocol::GetName(const MsgData& message) {
+    utils::JsonReader reader_;
+    Json::Value value;
+    
+    if(reader_.parse((char*)(message.data()), &value)) {
+
+        auto params = value.get(cParamsKey, Json::Value::null);
+        return GeStrFromParams(params, cNameKey);
+    }
+
+    return std::string();
```

Comment:
@VadymLuchko you can move the rest of impl to `GeStrFromParams`
```suggestion
        const auto json = GetParsedJson(message);
        return GetStrFromJson(json.get(cParamsKey, Json::Value::null), cNameKey);
```

Reply:
changed in 621f162d3fcd9ff3b9a74aa8bc83f4fa326ae133